driver-examples/osgi/src/test/java/com/datastax/driver/osgi/VersionProvider.java

```diff
@@ -25,11 +28,23 @@
  */
 public class VersionProvider {
 
+    private static final Pattern versionPattern = Pattern.compile(("(\\d+.\\d+\\.\\d+)(.*)"));
+
     private static String PROJECT_VERSION;
     static {
         String bundleName = Manifests.read("Bundle-SymbolicName");
         if (bundleName.equals("com.datastax.driver.osgi")) {
-            PROJECT_VERSION = Manifests.read("Bundle-Version").replaceAll("\\.SNAPSHOT", "-SNAPSHOT");
+            String bundleVersion = Manifests.read("Bundle-Version");
+            Matcher matcher = versionPattern.matcher(bundleVersion);
+            if(matcher.matches() && matcher.groupCount() == 2) {
+                String majorVersion = matcher.group(1);
+                // Replace all instances of '.' after the main version with '-' to properly
+                // resolve the correct version.
+                String rest = matcher.group(2).replaceAll("\\.", "-");
+                PROJECT_VERSION = majorVersion + rest;
+            } else {
+                PROJECT_VERSION = Manifests.read("Bundle-Version");
```

Comment:
The problem is that the `Bundle-Version` attribute replaces all dashes (-) with dots (.).  This reverts that so the driver jar can be resolved, unless I'm misunderstanding something.